packages/core/src/tracing/span.ts

```diff
@@ -297,7 +320,7 @@ export class Span implements SpanInterface {
    */
   public toContext(): SpanContext {
     return dropUndefinedKeys({
-      data: this.data,
+      data: this._getData(),
```

Comment:
Just to double check, I guess/hope it is not important that we never return `undefined` here? In contrast to `toJSON()` and `getTraceContext()`? I've aligned this here so that this always returns undefined if there is no data at all. cc @AbhiPrasad

Reply:
I think it's fine that we never return `undefined`

Reply:
but is it _also_ fine to return `undefined`? 😅 otherwise I can update this to `data: this._getData() || {}` but IMHO if we don't _need_ this we rather avoid it?

Reply:
We'll I'm gonna merge this anyway, I think it should be fine, otherwise we can adjust it later.